File: src/main/java/es/uji/sape/model/ProjectOffer.java
package es.uji.sape.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

import static es.uji.sape.model.Itinerary.SOFTWARE_ENGINEERING;
import static es.uji.sape.model.OfferState.PENDING;

@Data
@NoArgsConstructor
public final class ProjectOffer {
    private int id;
    private @NotNull Itinerary itinerary = SOFTWARE_ENGINEERING;
    private @NotNull String technologies = "";
    private @NotNull String objectives = "";
    private @NotNull OfferState state = PENDING;
    private int internshipOfferId;
}

File: src/main/kotlin/es/uji/sape/model/ProjectOffer.kt
package es.uji.sape.model

import es.uji.sape.model.Itinerary.SOFTWARE_ENGINEERING
import es.uji.sape.model.OfferState.PENDING

data class ProjectOffer(
    val id: Int = 0,
    val itinerary: Itinerary = SOFTWARE_ENGINEERING,
    val technologies: String = "",
    val objectives: String = "",
    val state: OfferState = PENDING,
    val internshipOfferId: Int = 0
)
